Add support for ssh-askpass

bapt opened the pull request:
Losely based on openssh code, add support for ssh-askpass programs
via the KP_ASKPASS env variable

It is useful when running kickpass without a real terminal

Commit:
Add support for ssh-askpass

Losely based on openssh code, add support for ssh-askpass programs
via the KP_ASKPASS env variable

It is useful when running kickpass without a real terminal

paulfariello commented:
Cool ! Thanks for the PR.
I've just merged feature/agent that add a password prompt callback.
It would be a better way to choose between classic prompt and askpass.
I'll work on it next week but if you have time before feel free to rebase your work ;)

bapt commented:
rebased

paulfariello committed:
Use ctx->password_cb everywhere

This way user can choose whatever method he wants for password prompt.

paulfariello committed:
Rename password_cb into password_prompt

password_cb is a callback but it's primarily for prompting password

paulfariello committed:
Choose askpass or readpass from main

askpass/readpass are set in password_prompt on ctx at kickpass startup.
Also review/rework a bit askpass.
